crates/storage-pg/migrations/20250225091000_dynamic_policy_data.sql

```diff
@@ -0,0 +1,11 @@
+-- Copyright 2025 New Vector Ltd.
+--
+-- SPDX-License-Identifier: AGPL-3.0-only
+-- Please see LICENSE in the repository root for full details.
+
+-- Add a table which stores the latest policy data
+CREATE TABLE IF NOT EXISTS policy_data (
+    policy_data_id UUID PRIMARY KEY,
```

Comment:
I don't particularly see why we bother to keep around multiple rows and track an UUID on them.

Separately, I could see it being desirable to have different policy data for different policies, so surprised there isn't a stringy key here.

Overall slightly unsure why this isn't 'just' a `TABLE policy_data ( key TEXT PRIMARY KEY NOT NULL, value JSONB NOT NULL )` to be honest! You probably have good reasons but would be nice to put them down

Reply:
I think the main motivation is being able to audit policy decisions? You get the `policy_data_id` logged every time it changes, so you can track what policy data was effective at that time

I also did that for consistency with other tables I guess 🤷 

Note that the policy is a single blob with multiple entrypoint. It shares the same `data` object for all entrypoints, so that's why there is no `key` in the schema here

Reply:
fair enough.

But we have this `prune` method; when does this get called? (I don't see any calls to it except in the tests for now)

Filling your disk because you're updating a large policy data blob frequently is also something to be avoided :-)

Reply:
Added a comment in 97d2b757c, and you're right, I've meant to add a job to regularly prune stale policy data, and failed to do so. Did that in 8581ca19e